src/Symfony/Component/PropertyAccess/Mapping/AttributeMetadata.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PropertyAccess\Mapping;
+
+/**
+ * {@inheritdoc}
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class AttributeMetadata implements AttributeMetadataInterface
```

Comment:
If we introduce new metadata classes (value objects), I propose to make them immutable like I've done in the PropertyInfo component (and API Platform v2): https://github.com/api-platform/core/blob/master/src/Metadata/Property/ItemMetadata.php